modules/SofaMiscTopology/SofaMiscTopology_test/TopologicalChangeProcessor_test.cpp

```diff
@@ -19,78 +19,379 @@
+    {
+        m_fileName = "IncisionTrianglesProcess.scn";
+    }
+
+    bool testTopologyChanges() override
+    {
+        Node::SPtr root = m_instance.root;
+
+        if (!root)
+        {
+            ADD_FAILURE() << "Error while loading the scene: " << m_fileName << std::endl;
+            return false;
+        }
+
+
+        Node::SPtr nodeTopo = root.get()->getChild("SquareGravity");
+        if (!nodeTopo)
+        {
+            ADD_FAILURE() << "Error 'SquareGravity' Node not found in scene: " << m_fileName << std::endl;
+            return false;
+        }
+
+        TriangleSetTopologyContainer* topoCon = dynamic_cast<TriangleSetTopologyContainer*>(nodeTopo->getMeshTopology());
+        if (topoCon == nullptr)
+        {
+            ADD_FAILURE() << "Error: TriangleSetTopologyContainer not found in 'SquareGravity' Node, in scene: " << m_fileName << std::endl;
+            return false;
+        }
+
+
+        // check topology at start
+        EXPECT_EQ(topoCon->getNbTriangles(), 1450);
+        EXPECT_EQ(topoCon->getNbEdges(), 2223);
+        EXPECT_EQ(topoCon->getNbPoints(), 774);
+
+        // to test incise animates the scene at least 1.2s
+        for (int i = 0; i < 50; i++)
+        {
+            m_instance.simulate(0.05);
+        }
+
+        EXPECT_EQ(topoCon->getNbTriangles(), 1677);
+        EXPECT_EQ(topoCon->getNbEdges(), 2704);
+        EXPECT_EQ(topoCon->getNbPoints(), 1026);
+
+        return true;
+    }
+};
+
+
+struct RemoveTriangleProcessor_test : TopologicalChangeProcessor_test
+{
+    RemoveTriangleProcessor_test() : TopologicalChangeProcessor_test()
+    {
+        m_fileName = "RemovingTrianglesProcess.scn";
+    }
+
+    bool testTopologyChanges() override
+    {
+        Node::SPtr root = m_instance.root;
+
+        if (!root)
+        {
+            ADD_FAILURE() << "Error while loading the scene: " << m_fileName << std::endl;
+            return false;
+        }
+
+
+        Node::SPtr nodeTopo = root.get()->getChild("SquareGravity");
+        if (!nodeTopo)
+        {
+            ADD_FAILURE() << "Error 'SquareGravity' Node not found in scene: " << m_fileName << std::endl;
+            return false;
+        }
+
+        TriangleSetTopologyContainer* topoCon = dynamic_cast<TriangleSetTopologyContainer*>(nodeTopo->getMeshTopology());
+        if (topoCon == nullptr)
+        {
+            ADD_FAILURE() << "Error: TriangleSetTopologyContainer not found in 'SquareGravity' Node, in scene: " << m_fileName << std::endl;
+            return false;
+        }
+
+
+        // check topology at start
+        EXPECT_EQ(topoCon->getNbTriangles(), 1450);
+        EXPECT_EQ(topoCon->getNbEdges(), 2223);
+        EXPECT_EQ(topoCon->getNbPoints(), 774);
+
+        //// to test incise animates the scene at least 1.2s
+        try {
+            for (int i = 0; i < 20; i++)
+            {
+                m_instance.simulate(0.01);
+            }
+}
+        catch (const std::exception& e) {
```

Comment:
I suppose those try{} catch{} comes from your previous tests to debug your code, should you remote those lines ?